src/server/qgsfcgiserverrequest.h

```diff
@@ -41,13 +41,22 @@ class SERVER_EXPORT QgsFcgiServerRequest: public QgsServerRequest
      */
     bool hasError() const { return mHasError; }
 
+    /**
+     * Returns the header value
+     * \param name of the header
+     * \return the header value or an empty string
+     * \since QGIS 3.20
+     */
+    QString header( const QString &name ) const override;
+
   private:
     void readData();
 
     // Log request info: print debug infos
     // about the request
     void printRequestInfos( const QUrl &url );
 
+    void fillServer( QUrl &url ) const;
```

Comment:
can you explain what exactly this method exactly does? I think we need a better name anyway.
Can `url` be a const ref?

Reply:
Comments added and renamed to `fillUrl`
the `url` can't be const because this method modify it :-)